ivadomed/transforms.py

```diff
@@ -1101,166 +838,88 @@ def __call__(self, sample):
         return sample
 
 
-class RandomTensorChannelShift(IMEDTransform):
+class RandomShiftIntensity(ImedTransform):
 
-    def __init__(self, shift_range):
+    def __init__(self, shift_range, prob=0.1):
         self.shift_range = shift_range
+        self.prob = prob
 
-    @staticmethod
-    def get_params(shift_range):
-        sampled_value = np.random.uniform(shift_range[0],
-                                          shift_range[1])
-        return sampled_value
-
-    @staticmethod
-    def sample_augment(input_data, params):
-        np_input_data = np.array(input_data)
-        np_input_data += params
-        input_data = Image.fromarray(np_input_data, mode='F')
-        return input_data
-
-    def __call__(self, sample):
-        input_data = sample['input']
-        params = self.get_params(self.shift_range)
-
-        if isinstance(input_data, list):
-            ret_input = [self.sample_augment(item, params) for item in input_data]
+    @multichannel_capable
+    def __call__(self, sample, metadata={}):
+        if np.random.random() < self.prob:
+            # Get random offset
+            offset = np.random.uniform(self.shift_range[0], self.shift_range[1])
         else:
-            ret_input = self.sample_augment(input_data, params)
+            offset = 0.0
 
-        rdict = {'input': ret_input}
+        # Update metadata
+        metadata['offset'] = offset
+        # Shift intensity
+        data = (sample + offset).astype(sample.dtype)
+        return data, metadata
 
-        sample.update(rdict)
-        return sample
+    @multichannel_capable
+    def undo_transform(self, sample, metadata={}):
+        assert 'offset' in metadata
+        # Get offset
+        offset = metadata['offset']
+        # Substract offset
+        data = (sample - offset).astype(sample.dtype)
+        return data, metadata
 
 
-class ElasticTransform(IMEDTransform):
-    "Elastic transform for 2D and 3D inputs"
+class ElasticTransform(ImedTransform):
+    """Elastic transform for 2D and 3D inputs."""
 
-    def __init__(self, alpha_range, sigma_range,
-                 p=0.5, labeled=True):
+    def __init__(self, alpha_range, sigma_range):
         self.alpha_range = alpha_range
         self.sigma_range = sigma_range
-        self.labeled = labeled
-        self.p = p
```

Comment:
Why remove p from ElasticTransform?

Reply:
I was annoying me when I was optimising alpha & sigma eheh Forgot to put it back! Many thanks :) Good spotting